src/telempostdaemon.c

```diff
@@ -417,17 +505,23 @@ static bool deliver_record(TelemPostDaemon *daemon, char *headers[], char *body)
         current_minute = tm_s->tm_min;
         /* Checks flags */
         bool record_check_passed = true;
+        bool record_to_json_passed = false;
         bool byte_check_passed = true;
         bool record_burst_enabled = burst_limit_enabled(daemon->record_burst_limit);
         bool byte_burst_enabled =  burst_limit_enabled(daemon->byte_burst_limit);
 
         /* Perform record and byte rate limiting checks */
         rate_limit_checks(daemon, &record_check_passed, &byte_check_passed);
 
+        /* Convert record headers and payload into json. */
+        json_str = record_to_json(headers, body);
+        record_to_json_passed = (json_str == NULL) ? false : true;
+
         /* Sends record if rate limiting is disabled, or all checks passed */
-        if (!daemon->rate_limit_enabled || (record_check_passed && byte_check_passed)) {
+        if (!daemon->rate_limit_enabled ||
+            (record_check_passed && byte_check_passed && record_to_json_passed)) {
```

Comment:
1) Line 518 evaluates record_to_json_passed to be either true or false. If it is false, that means there was no JSON record conversion and so no attempt is made to http-Tx any record.
Line 537 ret is not evaluated and ret remains false, because of ret initialization.

Reply:
My bigger point is that if the records can not be converted to JSON it is going to get stuck in spool. But, I think that's fine since this is the current behavior.